ambry-store/src/main/java/com/github/ambry/store/RemoteTokenTracker.java

```diff
@@ -69,7 +101,204 @@ void refreshPeerReplicaTokens() {
   /**
    * @return a snapshot of peer replica to token map.
    */
-  Map<String, FindToken> getPeerReplicaAndToken() {
+  Map<String, Pair<Long, FindToken>> getPeerReplicaAndToken() {
     return new HashMap<>(peerReplicaAndToken);
   }
+
+  /**
+   * @return the file path which persists the remote token.
+   */
+  protected String getPersistFilePath() {
+    return localReplica.getReplicaPath() + File.separator + REMOTE_TOKEN_FILE_NAME;
+  }
+
+  /**
+   * Provide a way to explicitly persist the token besides the background persistor
+   */
+  protected boolean persistToken() {
+    try {
+      persistor.persist();
+      return true;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Disable this {@link RemoteTokenTracker} and cancel the scheduled persistor.
+   */
+  @Override
+  public void close() {
+    persistor.cancel();
+  }
+
+  /**
+   * Start the background persistor.
+   */
+  public void start(int persistIntervalInSeconds) {
+    if (persistIntervalInSeconds > 0) {
+      if (this.scheduler != null) {
+        this.scheduler.scheduleAtFixedRate(persistor, 0, persistIntervalInSeconds, TimeUnit.SECONDS);
+        logger.info("flush peer's remote token every {} seconds", persistIntervalInSeconds);
+      } else {
+        logger.error("scheduler is null, couldn't persist peer's remote token.");
+      }
+    }
+  }
+
+  /**
+   * Runner that persist the remote peers and their tokens.
+   */
+  private class RemoteTokenPersistor implements Runnable {
+    private volatile boolean cancelled = false;
+
+    @Override
+    public void run() {
+      if (!cancelled) {
```

Comment:
we don't need a cancelled boolean here, we can let the schedule return a ScheduledFuture and cancel the future when closing the RemoteTokeneTracker.

Reply:
Done.